File: Code/ConnectFour.py
N=6

Board = [
[1,0,0,0,0,2],
[0,2,0,0,2,0],
[0,0,1,2,0,0],
[0,0,2,1,0,0],
[0,2,0,1,0,0],
[0,2,0,1,0,0]
]

def isWinner( x, y, p):
    countRows = 0
    countCols = 0
    countUpDiag = 0
    countDownDiag = 0

    for i in xrange(4):
        if x+i < N and p ==  Board[x+i][y]: countRows += 1
        if y+i < N and p ==  Board[x][y+i]: countCols += 1
        if x+i < N and y+i < N and p ==  Board[x+i][y+i]: countUpDiag += 1
        if x-i >= 0 and y+i < N and p ==  Board[x-i][y+i]: countDownDiag += 1

    if countRows == 4 or countCols == 4 or countUpDiag == 4 or countDownDiag == 4:
        return p

    return 0

def getWinner():
   #val = 0
   for i, j  in[(i,j) for i in range(N) for j in range(N)]:
          val = isWinner(i,j,Board[x][y])
          if val:
              return val
   return 0

def main ():
  print "winner: ",  getWinner()

if __name__ == "__main__":
    main()

File: Code/ConnectFour.cpp
/******************************************************************************

Welcome to GDB Online.
GDB online is an online compiler and debugger tool for C, C++, Python, Java, PHP.
Code, Compile, Run and Debug online from anywhere in world.

*******************************************************************************/
#include <iostream>

using namespace std;

const int N = 5;

int Board[N][N] = {
  {0,0,0,0,0},
  {0,0,0,0,2},
  {0,0,0,2,0},
  {0,0,2,0,0},
  {0,2,0,0,0}
};

int isWinner(int x, int y, int color)
{
   
    int countRows = 0;
    int countCols = 0;
    int countDiagUp = 0;
    int countDiagDown = 0;

    for(int i = 0; i < 4 ; i++) {
        if(x+i<N && color ==  Board[x+i][y]) countRows++;
        if(y+i<N && color ==  Board[x][y+i]) countCols++;
        if(x+i<N && y+i<N && color ==  Board[x+i][y+i]) countDiagUp++;
        if(x-i >= 0 && y+i<N && color ==  Board[x-i][y+i]) countDiagDown++;
    }
    if( countRows == 4 || 
      countCols == 4 ||
      countDiagUp == 4 ||
      countDiagDown == 4 )
       return color;
 
    return 0;
}


int getWinner()
{
    int val;
    
    for( int i = 0;i<N;i++)
       for( int j = 0;j<N;j++)
         if (val = isWinner(i,j, Board[i][j]) )
           return val;

    return 0;
} 

int main ()
{
  cout << "winner: " << getWinner ();

  return 0;
}